test/fw/ptl/utils/plugins/ptl_test_runner.py

```diff
@@ -599,6 +685,13 @@ def startTest(self, test):
             self.__failed_tc_count_msg = True
             raise TCThresholdReached
         timeout = self.__get_timeout(test)
+        rv = self.__are_requirements_matching(self.param_dict, test)
```

Comment:
The decorator isn't really doing much besides setting REQUIREMENTS_KEY, the actual checking for requirements is happening here as part of startTest(). Is there no way to move this check to the decorator itself? The whole point of a decorator is to run it before starting a test.

Reply:
This decorator is supported at test suite level and test case level. I found it difficult to get both test suite level and test case level requirements data together in the same call inside the decorator function, hence could not compute the effective requirements for a given test case context.

Regarding the other data which is param data of benchpress, in order to get that inside decorator it could be got by a function defined (in pbs_testsuite.py) to set a PBSTestSuite class level macro to the merged param data (-p and paramfile); and call this function from pbs_benchpress after importing. 

So, I preferred to have this current code flow where the merged param data is already available in runner plugin and the requirements data that is set on test object could be captured at different levels i.e. from test method (current test name) and its class which is the test suite. (code lines 640 - 646 in ptl_test_runner.py)